### PR TITLE
feat(replays): Replay tabs current position styles

### LINKED ISSUE #38947: Replay Details: border-bottom indicator to signal where a user is in the replay video for Network, Console and DOM Events tab
We want to add a purple line to distinguish between rows that are 'in the past' vs. 'coming up next', corresponding to where a user is at in the replay video for the Network, Console, and DOM Events tab. This is a key usability feature.

Image

See the [Figma](https://www.figma.com/file/OPFnT04z0uy8BEUms9OU4f/Specs%3A-Details-v3?node-id=142%3A7844).

Internal feedback quote:

> another related thing that would be super helpful is to have an indicator somewhere on the network tab based on the current time which of these have happened yet and which haven’t. I’m trying to do some debugging to understand what part of the video corresponds to which network response and that’s hard right now because I have to cross reference the timestamps instead of getting an indicator

This is an implementation ticket from the design ticket [#38620](https://github.com/getsentry/sentry/issues/38620#issuecomment-1248593395)

### PR DESCRIPTION
**Pull request by danecando**
### Changes
- Changed `getBreadcrumbs` to be generic and renamed it to `getReplayEvent`
- Added replay position styles to network tab items
- Added replay position styles to dom event tab items

### Test Cases
I have tested the functionality in the following ways

#### DOM Mutations
- [x]  Underline is below the last item of the videos current playback position
- [x] Items that are after the current playback have the muted styles applied (hasOccurred false)
- [x] When filters are applied, the current item and hasOccurred styles are only applied to the items that are not filtered
- [x] Clicking on a timestamp of a mutation will set the playback time and the current item underline and hasOccurred styles are correct

#### Network Tab
- [x] During playback underline is below the latest item relative to the current playback position
- [x] Items that are after the current playback have muted styled applied (hasOccured false)
- [x] Current item does not operate on filtered network items so the current underline will always be relative to the last item that is currently visible
- [x] Clicking on a timestamp of a network item will set the playback time and the current item underline and hasOccurred styles are correct
- [x] Sorting network items by any column that is not the timestamp disabled current underline and hasOccurred styles
- [x] Sorting by reverse time applies the same styles but in backwards order with the underline above the last item

### Notes
Network tab can be sorted by columns other than timestamp which breaks down the play position styles since they will no longer be ordered by time. I have set the styles to default when sorted by any column other then timestamp.

The color change for dom event item icons that have not yet occurred adds transparency so you can see the connector behind it a little bit. The lighter purple values add transparency, in the new mocks a blue color is used (So I assume this will not be an issue when we update the styles in a future PR)

Closes #38947 

### Demo
 
https://user-images.githubusercontent.com/3721977/191578599-cb1679cd-5042-45ed-b0e5-f004ade498b8.mp4

Screen Shot 2022-09-21 at 2 08 23 PM
Screen Shot 2022-09-21 at 2 08 30 PM

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/replays/detail/domMutations/index.tsx [hunk 1]
```diff
@@ -215,39 +262,13 @@ const MutationListItem = styled('li')`
...
-  width: 24px;
-  min-width: 24px;
-  height: 24px;
-  border-radius: 50%;
-  color: ${p => p.theme.white};
-  background: ${p => p.theme[p.color] ?? p.color};
-  box-shadow: ${p => p.theme.dropShadowLightest};
-  z-index: 2;
+  ${MutationDetailsContainer} {
+    opacity: ${p => (p.hasOccurred ? 1 : 0.5)};
```
Comment (ryan953) [inline static/app/views/replays/detail/domMutations/index.tsx:267]: is opacity the only way to do this? how will we solve the problem when the icon colors get updated?

Comment (ryan953) [reply static/app/views/replays/detail/domMutations/index.tsx:267]: the existing console implementation does not use opacity. they should be using the same techniques afaik.

**Review by ryan953 — CHANGES_REQUESTED** [review]
The colors look wrong on the network table:

When I click the timestamp of an item in the network table the line goes on top of that item:
Screen Shot 2022-09-21 at 2 18 28 PM

File: static/app/views/replays/detail/domMutations/index.tsx [hunk 2]
```diff
@@ -48,7 +53,10 @@ function DomMutations({replay}: Props) {
     setSearchTerm,
   } = useDomFilters({actions});
 
-  const startTimestampMs = replay.getReplay().startedAt.getTime();
+  const currentDomMutation = getPrevReplayEvent({
+    items: items.map(mutation => mutation.crumb),
+    targetTimestampMs: startTimestampMs + currentTime,
+  });
```
Comment (ryan953) [inline static/app/views/replays/detail/domMutations/index.tsx:59]: just comparing this to the console table, when `getPrevReplayEvent` is called.. i can see in the console table we're setting `allowExact: true, allowEqual: true,` and here it's not set.

File: static/app/views/replays/detail/domMutations/index.tsx (hunk 1 above)
Comment (danecando) [reply static/app/views/replays/detail/domMutations/index.tsx]: The lighter colored icon is itself opaque. So we will need a lighter icon color that doesn't have a lesser opacity

**Review by ryan953 — APPROVED** [review]

**Comment by ryan953** [comment]
There's a double border in the Network table when the timestamp column is sorted desc (end of video on top)
Screen Shot 2022-09-22 at 9 43 33 AM

We'll can do a follow up PR for that, I don't want to wait on CI for the rest of this.

**Comment by ryan953** [comment]
In the network table the gray font color is taking precedence over red:
Screen Shot 2022-09-22 at 9 46 50 AM
Screen Shot 2022-09-22 at 9 46 55 AM

I think this is the wrong way around, when i load up the page I don't get that color to attract me to the problems. The red could be faded or something.

Again, this will be a fast followup PR to fix.

**Pull request merged**